flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/sink/TopicSelector.java

```diff
@@ -18,14 +18,12 @@
 package org.apache.flink.connector.kafka.sink;
 
 import org.apache.flink.annotation.PublicEvolving;
-
-import java.io.Serializable;
-import java.util.function.Function;
+import org.apache.flink.util.function.SerializableFunction;
 
 /**
  * Selects a topic for the incoming record.
  *
  * @param <IN> type of the incoming record
  */
 @PublicEvolving
-public interface TopicSelector<IN> extends Function<IN, String>, Serializable {}
+public interface TopicSelector<IN> extends SerializableFunction<IN, String> {}
```

Comment:
This is a probably breaking change (just double check if user code is affected or not). Since this PR is aimed as a bugfix to 1.14.0, I'd remove it for now. Please also double-check the other migrations.

In general, I think that `TopicSelector` should not have inherited from `Function` to begin with. :/

Reply:
Good catch, yeah it breaks